homeassistant/components/tuya/climate.py

```diff
@@ -0,0 +1,464 @@
+                {"code": DPCODE_SWITCH_VERTICAL, "value": False},
+                {"code": DPCODE_SWITCH_HORIZONTAL, "value": False},
+            ]
+
+        self._send_command(commands)
+
+    def set_temperature(self, **kwargs: Any) -> None:
+        """Set new target temperature."""
+        _LOGGER.debug("climate temp-> %s", kwargs)
+        code = DPCODE_TEMP_SET if self.is_celsius() else DPCODE_TEMP_SET_F
+        self._send_command(
+            [
+                {
+                    "code": code,
+                    "value": int(
+                        kwargs["temperature"] * (10 ** self.get_temp_set_scale())
+                    ),
+                }
+            ]
+        )
+
+    def is_celsius(self) -> bool:
+        if (
+            self.dp_temp_unit in self.tuya_device.status
+            and self.tuya_device.status.get(self.dp_temp_unit).lower() == "c"
+        ):
+            return True
+        if (
+            DPCODE_TEMP_SET in self.tuya_device.status
+            or DPCODE_TEMP_CURRENT in self.tuya_device.status
+        ):
+            return True
+        return False
+
+    @property
+    def temperature_unit(self) -> str:
+        """Return true if fan is on."""
+        if self.is_celsius():
+            return TEMP_CELSIUS
+        return TEMP_FAHRENHEIT
+
+    @property
+    def current_temperature(self) -> float | None:
+        """Return the current temperature."""
+        if (
+            DPCODE_TEMP_CURRENT not in self.tuya_device.status
+            and DPCODE_TEMP_CURRENT_F not in self.tuya_device.status
+        ):
+            return None
+
+        if self.is_celsius():
+            return (
+                self.tuya_device.status.get(DPCODE_TEMP_CURRENT, 0)
+                * 1.0
+                / (10 ** self.get_temp_current_scale())
+            )
+        return (
+            self.tuya_device.status.get(DPCODE_TEMP_CURRENT_F, 0)
+            * 1.0
+            / (10 ** self.get_temp_current_scale())
+        )
+
+    @property
+    def current_humidity(self) -> int:
+        """Return the current humidity."""
+        return int(self.tuya_device.status.get(DPCODE_HUMIDITY_CURRENT, 0))
+
+    @property
+    def target_temperature(self) -> float:
+        """Return the temperature currently set to be reached."""
+        return (
+            self.tuya_device.status.get(DPCODE_TEMP_SET, 0)
+            * 1.0
+            / (10 ** self.get_temp_set_scale())
+        )
+
+    @property
+    def max_temp(self) -> float:
+        """Return the maximum temperature."""
+        return self.target_temperature_high
+
+    @property
+    def min_temp(self) -> float:
+        """Return the minimum temperature."""
+        return self.target_temperature_low
+
+    @property
+    def target_temperature_high(self) -> float:
+        """Return the upper bound target temperature."""
+        if self.is_celsius():
+            if DPCODE_TEMP_SET not in self.tuya_device.function:
+                return 0
```

Comment:
Please adjust all temperature properties to return None to represent unknown or missing value.

Reply:
fixed